testing/testcafe/tests/dataGrid/editing.ts

```diff
@@ -2125,24 +2126,114 @@ test('Cells should be focused correctly on click when cell editing mode is used
 });
 
 // T1130497
-test('The first cell of the last row should be focused when newRowPosition = last and editing.mode = cell', async (t) => {
-  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
-  const dataGrid = new DataGrid('#container');
-  const headerPanel = dataGrid.getHeaderPanel();
-
-  await t
-    .click(headerPanel.getAddRowButton())
-    .expect(dataGrid.getDataRow(3).isInserted).ok('row is inserted')
-    .expect(await takeScreenshot('grid-cell-edit-mode-and-new-row-position-last.png', dataGrid.element))
-    .ok()
-    .expect(compareResults.isValid())
-    .ok(compareResults.errorMessages());
-}).before(async () => createWidget('dxDataGrid', {
-  dataSource: [{ name: 'AaAaA', value: 1 }, { name: 'aAaAa', value: 2 }, { name: 'BbBb', value: 3 }],
-  editing: {
-    mode: 'cell',
-    allowUpdating: true,
-    allowAdding: true,
+[
+  {
+    newRowPosition: 'first',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
     newRowPosition: 'last',
+    insertedRowNumber: 20,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'pageBottom',
+    insertedRowNumber: 20,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'pageTop',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'pageBottom',
+    insertedRowNumber: 8,
+    scrollMode: 'virtual',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'pageTop',
+    insertedRowNumber: 0,
+    scrollMode: 'virtual',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 8,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 13,
+    scrollMode: 'standard',
+    scrollTop: 162,
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 5,
+    scrollMode: 'standard',
+    scrollTop: 162,
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 8,
+    scrollMode: 'virtual',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 13,
+    scrollMode: 'virtual',
+    scrollTop: 162,
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 0,
+    scrollMode: 'virtual',
+    scrollTop: 0,
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 5,
+    scrollMode: 'virtual',
+    scrollTop: 162,
   },
-}));
+].forEach((testCase) => {
+  test(`The first cell of the new row should be focused when
+      newRowPosition = ${testCase.newRowPosition}
+      and editing.mode = cell
+      and ${testCase.scrollMode} scroll mode
+      and scrollTop is ${testCase.scrollTop}`, async (t) => {
+    const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
```

Comment:
For a single screenshot comparison, you can use:
.expect(await compareScreenshot(t, 'pager-full-allpages.png'))

More info:
https://www.npmjs.com/package/devextreme-screenshot-comparer